example/spec/asyncOperations/asyncOperationRunnerSpec.js

```diff
@@ -0,0 +1,392 @@
+      expect(dynamoDbItem.status.S).toEqual('RUNNER_FAILED');
+    });
+
+    it('updates the output field in DynamoDB', async () => {
+      const parsedOutput = JSON.parse(dynamoDbItem.output.S);
+
+      expect(parsedOutput.message).toBe(`Failed to fetch ${payloadUrl}: The specified key does not exist.`);
+    });
+
+    it('updates the updatedAt field in DynamoDB', async () => {
+      expect(dynamoDbItem.updatedAt.N).toBeGreaterThan(dynamoDbItem.createdAt.N);
+    });
+  });
+
+  describe('with a non-JSON payload', () => {
+    let asyncOperationId;
+    let taskArn;
+    let dynamoDbItem;
+    let payloadKey;
+
+    beforeAll(async () => {
+      asyncOperationId = uuidv4();
+
+      // Upload the payload
+      payloadKey = `${config.stackName}/integration-tests/payloads/${asyncOperationId}.json`;
+      await s3().putObject({
+        Bucket: config.bucket,
+        Key: payloadKey,
+        Body: 'invalid JSON'
+      }).promise();
+
+      await asyncOperationModel.create({
+        id: asyncOperationId,
+        taskArn: randomString(),
+        status: 'RUNNING'
+      });
+
+      const runTaskResponse = await ecs().runTask({
+        cluster,
+        taskDefinition: asyncOperationTaskDefinition,
+        launchType: 'EC2',
+        overrides: {
+          containerOverrides: [
+            {
+              name: 'AsyncOperation',
+              environment: [
+                { name: 'asyncOperationId', value: asyncOperationId },
+                { name: 'asyncOperationsTable', value: config.AsyncOperationsTable },
+                { name: 'lambdaName', value: successFunctionName },
+                { name: 'payloadUrl', value: `s3://${config.bucket}/${payloadKey}` }
+              ]
+            }
+          ]
+        }
+      }).promise();
+
+      taskArn = runTaskResponse.tasks[0].taskArn;
+
+      await ecs().waitFor(
+        'tasksStopped',
+        {
+          cluster,
+          tasks: [taskArn]
+        }
+      ).promise();
+
+      dynamoDbItem = await waitForAsyncOperationStatus({
+        TableName: config.AsyncOperationsTable,
+        id: asyncOperationId,
+        status: 'TASK_FAILED'
+      });
+    });
+
+    it('updates the status field in DynamoDB to "TASK_FAILED"', async () => {
+      expect(dynamoDbItem.status.S).toEqual('TASK_FAILED');
+    });
+
+    it('updates the output field in DynamoDB', async () => {
+      const parsedOutput = JSON.parse(dynamoDbItem.output.S);
+
+      expect(parsedOutput.message).toContain('Unable to parse payload:');
+    });
+
+    it('updates the updatedAt field in DynamoDB', async () => {
+      expect(dynamoDbItem.updatedAt.N).toBeGreaterThan(dynamoDbItem.createdAt.N);
+    });
+
+    afterAll(() => s3().deleteObject({ Bucket: config.bucket, Key: payloadKey }).promise());
+  });
+
+  describe('executing a successful lambda function', () => {
+    let asyncOperationId;
+    let taskArn;
+    let dynamoDbItem;
+    let payloadKey;
+
+    beforeAll(async () => {
```

Comment:
I'm curious why this is testing the pieces inside start() instead of just calling asyncopertionsmodel.start(), and it looks like because we're not passing in params.id anymore so you didn't want to mock uuidv4()?  Or some other thing that I don't understand?  It seems like if something changes in start, these will all have to change?

Reply:
This isn't using `asyncOperationsModel.start()` because we need to be able to create failure conditions.  (Non-existent payload, invalid JSON in the payload, etc).